content_sync/tasks.py

```diff
@@ -225,6 +225,25 @@ def sync_website_content(website_name: str):
         backend.sync_all_content_to_backend()
 
 
+@app.task(acks_late=True)
+def update_mass_build_pipelines(website: Website, version: str):
+    """Update the mass-build-sites pipeline definitions"""
+    if settings.CONTENT_SYNC_PIPELINE_BACKEND:
+        publish_date_field = (
+            "publish_date" if version == VERSION_LIVE else "draft_publish_date"
+        )
+        publish_date = getattr(website, publish_date_field)
+        if not publish_date:
+            pipeline = api.get_mass_build_sites_pipeline(
+                version, offline=False, prefix="", starter=""
+            )
+            pipeline.upsert_pipeline()
```

Comment:
Would anything bad happen if this pipeline is updated while running?  For instance someone does a fresh publish of a new site while a mass build is ongoing.  Does the concourse pipeline run keep processing, using the old definition, until complete?

Reply:
This is also a good question. I believe what would happen is the current batch job would finish as-is, then once the next batch job started it would be using the new definition.

Reply:
This doesn't seem to be getting called when I publish a website for the 1st time.  Ditto for a brand new site when I publish to draft. No mass build pipelines are generated.  I only see it getting called in mgmt commands and unit tests.  Shouldn't it be added [here](https://github.com/mitodl/ocw-studio/pull/1990/files#diff-4ac42200ced3c3c941d5c0c61cd8643a63f30994e580d567033dfea5b56da7adR165-R166)?

Reply:
I just added a point to the testing instructions I'm realizing I forgot about, which involves setting content_sync.tasks.update_mass_build_pipelines on PREPUBLISH_ACTIONS in order to ensure the function is run when a site is published.

Reply:
Now that you mention it though, I think it would be better to add the function call here rather than depending on the function being called as a prepublish action.